src/promptflow/tests/sdk_cli_test/unittests/test_utils.py

```diff
@@ -202,18 +202,20 @@ def test_concurrent_execution_of_refresh_connections_dir(self, concurrent_count)
         for thread in threads:
             thread.join()
 
-    @pytest.mark.parametrize("concurrent_count", [1, 2, 4, 8])
-    def test_concurrent_hint_for_update(self, concurrent_count):
-        from concurrent.futures import ThreadPoolExecutor
+    def test_concurrent_hint_for_update(self):
+        import time
         with patch('promptflow._utils.version_hint_utils.datetime') as mock_datetime:
             mock_datetime.datetime.now.return_value = datetime.datetime.now()
             mock_datetime.datetime.strptime.return_value = datetime.datetime.now() - datetime.timedelta(days=8)
             mock_datetime.timedelta.return_value = datetime.timedelta(days=7)
             hint_for_update()
-            with ThreadPoolExecutor(max_workers=concurrent_count) as pool:
-                pool.submit(check_latest_version)
-
+            prev_time = time.perf_counter()
+            thread = threading.Thread(target=check_latest_version, daemon=True)
```

Comment:
Can the test be decorated with the "monitor function" to fully run the internal logic of the entire monitor function?
Control "activity name" to go to the thread logic.

Reply:
Updated.
Since time here can't judge whether child thread won't block main thread (even child thread isn't a Deamon thread, time also a small number in main thread, but actually main thread waits for child thread), so use mock_check_latest_version here to judge "LAST_CHECK_TIME" key.

Reply:
updated: 
Since run test concurrently in CI, ` LAST_CHECK_TIME ` may exists in cache file by other tests. 
And although other concurrent tests may also change the value, but we have 7 days check. So we can use the value of
`LAST_CHECK_TIME` to ensure time is not set by `test_concurrent_hint_for_update`. I think the probability of time collision is very small.